### PR TITLE
Refactor couchdb scaler

### PR DESCRIPTION
**Pull request by rickbrouwer**
- Refactor the scaler to match the new metadata guidelines.

### Checklist

- [X] I have verified that my change is according to the [deprecations & breaking changes policy](https://github.com/kedacore/governance/blob/main/DEPRECATIONS.md)
- [X] Tests have been added
- [X] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))

Related: [5797](https://github.com/kedacore/keda/pull/5797)

### CODE REVIEW

File: pkg/scalers/couchdb_scaler.go [hunk 1]
```diff
@@ -19,216 +18,173 @@ import (
...
-		meta.queryValue = queryValue
-	} else {
-		if config.AsMetricSource {
-			meta.queryValue = 0
-		} else {
-			return nil, "", fmt.Errorf("no queryValue given")
-		}
+	err = meta.Validate()
+	if err != nil {
+		return meta, err
```
Comment (wozniakjan) [inline pkg/scalers/couchdb_scaler.go:120]: shouldn't be necessary to call explicitly as it should be called implicitly as part of `TypedConfig()` invocation

https://github.com/kedacore/keda/blob/b2ce95daaa8d5f8b4842561c594b8767a7194bd4/pkg/scalers/scalersconfig/typed_config.go#L184-L188

**Review by wozniakjan — APPROVED** [review]
lgtm, thank you!

minor nit below

File: pkg/scalers/couchdb_scaler.go (hunk 1 above)
Comment (rickbrouwer) [reply pkg/scalers/couchdb_scaler.go]: Removed, thanks for the feedback!

Comment (rickbrouwer) [reply pkg/scalers/couchdb_scaler.go]: Hi @wozniakjan . Can you re-rerun the `/run-e2e couchdb`?

**Review by JorTurFer — APPROVED** [review]

**Pull request merged**